airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -493,3 +496,39 @@ def get_container(self, container_list):
                     'container "name" must be provided in "containerOverrides" configuration'
                 )
         raise KeyError(f"No such container found by container name: {self.container_name}")
+
+    def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[TaskInstance]:
+        """
+        Adopt task instances which have an external_executor_id (the ECS task ARN).
```

Comment:
I might have missed this, but how is the task instance getting the `external_executor_id` attached to it?

Reply:
[here](https://github.com/apache/airflow/compare/main...aws-mwaa:upstream-to-airflow:ferruzzi/ecs-executor/adoption-agency#diff-a72b32930fa3a0eb853fcdf4ff4692a096762f7149da7e8cdbb5965c5c569607R398) when the task is first queued

Reply:
Ah nice, thanks!